frontend/scripts/react-components/shared/dropdown/dropdown.component.jsx

```diff
@@ -6,7 +6,32 @@ import cx from 'classnames';
 import './dropdown.scss';
 
 function Dropdown(props) {
-  function renderItem(item, index, getItemProps) {
+  const listItemRef = useRef(null);
+  const [listHeight, updateListHeigh] = useState(null);
+  useEffect(() => {
+    if (listItemRef.current && listHeight === null && props.options && props.options.length > 0) {
+      const { height } = listItemRef.current.getBoundingClientRect();
+      const optionsLength = props.showSelected ? props.options.length : props.options.length - 1;
+      if (height > 0 && height * optionsLength > Dropdown.DEFAULT_MAX_LIST_HEIGHT) {
+        const newListHeight = height * 6 - height / 2;
+        updateListHeigh(newListHeight);
+      }
+    }
+  });
```

Comment:
Reading about hooks maybe we can limit the useEffect calls adding a second [listItemRef] attribute. We only want to trigger it the first time,  if it changes from null to the reference, don't we?: https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
Ok, reading a bit more we would have to include all the variables used in the scope so maybe is not helpful

Reply:
A way of only executing once is to pass an empty array, this would be equivalent to `componentDidMount` however I think we need to run it at least twice so we’re screwed there.

You should only add as dependencies the values that cause a re-render so only state vars and props. Adding a ref is not recommended.

Because of the if statement this should only call the function once.